subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/logging/GroupedTaskFixture.java

```diff
@@ -65,11 +67,59 @@ public boolean isSatisfiedBy(String string) {
         return join("\n", nonEmptyOutputs);
     }
 
+    @Override
+    public String toString() {
+        return "Output for task: " + taskName + (null != taskOutcome ? " (" + taskOutcome + ")" : "");
+    }
+
+    @Deprecated // Use assertOutputContainsLines(String) instead
     public GroupedTaskFixture assertOutputContains(String... text) {
```

Comment:
I was thinking something a little simpler than having our own potential match/line searcher and it would apply everywhere (so we wouldn't need to deprecate or migrate to a different method).

The output here is always going to be relatively small with a known context because it's going to be the output for a single task.

This is closer to what I was thinking, I'm not sure about the hardcoded newlines here or context:
```
    public GroupedTaskFixture assertOutputContains(String... text) {
        String output = getOutput();
        for (String s : text) {
            assert output.contains(s) : "could not find requested text in task output, unified diff\n" + Joiner.on('\n').join(DiffUtilsAccess.generateUnifiedDiff(Arrays.asList(output.split("\n")), Arrays.asList(s.split("\n")), 20));
        }
        return this;
    }
```

Here's what that looks like by breaking a test:
https://ge.gradle.org/s/eopywuzltoxrc/tests/:diagnostics:embeddedIntegTest/org.gradle.api.tasks.diagnostics.OutgoingVariantsReportTaskIntegrationTest/if%20single%20outgoing%20variant%20present%20with%20attributes,%20task%20reports%20it%20and%20them?anchor=e30&focused-exception-line=0-2&top-execution=1

You can see the unified diff without the power assertion:
https://ge.gradle.org/s/eopywuzltoxrc/tests/:diagnostics:embeddedIntegTest/org.gradle.api.tasks.diagnostics.OutgoingVariantsReportTaskIntegrationTest/if%20single%20outgoing%20variant%20present%20with%20attributes,%20task%20reports%20it%20and%20them?anchor=e30&focused-exception-line=0-1-0&top-execution=1

This doesn't look too bad in the build scan, but locally we still get the failure twice in the output. I think this is coming from IntelliJ somehow.

Reply:
We discussed output formats, and want to see if there are any other strong opinions on the format.  It seems different types of comparisons might benefit from different formats, and leaving a "menu" of formatting options in place might be a good solution.

The probably common case of blocks completely absent from one side of the comparison is something the unified diff handles much better, for instance; and the case of repititive, slightly different blocks where the unified diff will only attempt to highlight a single possible location of the match is something it does more poorly.  Depending on usage, lines numbers may or may not be useful, depending on content, the unified diff's symbology might be less readable, etc.

Maybe the "Tom diff" can be improved later to handle mulitple missing blocks better, but as long as the unified diff is available for use in this case, this is a very low priority.